### DISCUSSION TITLE
Do not expand on all possible function syntaxes in syntax boxes

### DISCUSSION THREAD

**Discussion by Josh-Cena** [mdn/mdn-community#356] [Content]
Consider this Array method syntax box:

```js
// Arrow function
map((element) => { /* … */ })
map((element, index) => { /* … */ })
map((element, index, array) => { /* … */ })

// Callback function
map(callbackFn)
map(callbackFn, thisArg)

// Inline callback function
map(function (element) { /* … */ })
map(function (element, index) { /* … */ })
map(function (element, index, array) { /* … */ })
map(function (element, index, array) { /* … */ }, thisArg)
```

This is extremely long, repetitive, and hard to maintain. Furthermore, we don't use this pattern in other places where a callback is accepted:

```js
new Promise(executor)
```

```js
JSON.stringify(value)
JSON.stringify(value, replacer)
JSON.stringify(value, replacer, space)
```

```js
replace(pattern, replacement)
```

Is it possible that we stop expanding all callback syntaxes, but instead use parameter descriptions?

````md
```js-online
map(callbackFn)
map(callbackFn, thisArg)
```

### Parameters

- `callbackFn`

  - : A function to execute for each element in the array. Its return value is added as a single element in the new array.

    The function is called with the following arguments:

    - `element`
      - : The current element being processed in the array.
    - `index`
      - : The index of the current element being processed in the array.
    - `array`
      - : The array `map()` was called upon.
````

This needs updates to the syntax guideline.

cc @mdn/yari-content-javascript

**Comment by teoli2003** [comment]
This sounds good to me.

**Comment by Elchi3** [comment]
I like your proposal. I originally worked this out when changing the syntax boxes in 2021, but whenever I looked at the callbacks specifically I wasn't quite happy with it. I wanted to make this beginner friendly but this part indeed got long and repetitive.

Context from 2021:
https://github.com/openwebdocs/project/issues/26
https://github.com/mdn/content/pull/4136
And especially https://github.com/mdn/content/pull/4249

Happy to review a PR against the guidelines and to use parameters as you propose. https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Syntax_sections#callback_syntax_blocks

**Comment by OnkarRuikar** [comment]
- Can we close this? as the guidelines have been updated https://github.com/mdn/content/pull/25752

Was this issue raised for only JavaScript? or the entire repo? API pages are still using the old style e.g. `Web/API/Headers/forEach`. Do we have to update rest of the repo as well?

**Reply by Josh-Cena** [reply]
I think we should update the web API pages as well. Maybe someone could do it?

**Reply by teoli2003** [reply]
https://github.com/mdn/content/pull/26121 landed. Thanks to @Elchi3 !

**Reply by OnkarRuikar** [reply]
A few still remain e.g.
> Web/JavaScript/Reference/Global_Objects/Array/toSorted
> Web/JavaScript/Reference/Global_Objects/TypedArray/toSorted

It's hard to find the rest of them as the syntax sections are not uniform. May be one day we'll get them all.

**Reply by Josh-Cena** [reply]
Thanks @OnkarRuikar, those were created by race condition (the PR was created before the update and merged after the update). Welcoming PRs.
